go/vaas/pkg/vaas/vaas.go

```diff
@@ -0,0 +1,323 @@
+package vaas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+
+	BroadcastChannel "vaas/pkg/broadcast_channel"
+	"vaas/pkg/hash"
+	msg "vaas/pkg/messages"
+	"vaas/pkg/options"
+
+	"github.com/gorilla/websocket"
+)
+
+const TIMEOUT = 60
+
+type Vaas struct {
+	sessionId           string
+	websocketConnection *websocket.Conn
+	broadcastChannel    *BroadcastChannel.BroadcastChannel[msg.VerdictResponse]
+	requestChannel      chan msg.IVerdictRequest
+	responseChannel     chan msg.VerdictResponse
+	Url                 url.URL `default:"wss://gateway-vaas.gdatasecurity.de"`
+	options             options.VaasOptions
+	Ctx                 context.Context
```

Comment:
You should not store a context in a struct 
https://go.dev/blog/context-and-structs